utils/node_local_test.py

```diff
@@ -114,6 +117,25 @@ def set_args(self, args):
     def __getitem__(self, key):
         return self.bc[key]
 
+    def compress_file(self, filename):
+        """Compress a file using bz2 for space reasons
+
+        Launch a bzip2 process in the background as this is time consuming, and each time
+        a new process is launched then reap any previous ones which have completed.
```

Comment:
Is it possible/realistic for there to be too many running subprocesses? Wondering if we need a `multiprocessing.Pool`

Reply:
not realistic I'd have thought.  we only have a few thousand log files which are mostly quite small and if we really got that far behind the python process itself would probably slow down.